### PR TITLE
Mobile theme tweaks

### LINKED ISSUE #9634: Theme feedback
Collecting some feedback on the beta themes here, starting with my own but we can add user feedback as we get it.

### Pyramid Fleet (Dark Theme)

* The red key color feels like too much
* Having the store header be transparent (seeing items behind it) is distracting and makes the header harder to parse.
* Black polaroids for masterworks feels weird and reduces their visibility rather than highlighting them.

cc @guise

### PR DESCRIPTION
**Pull request by guise**
Mobile theme fixes for #9634 

Added a variable to control the mobile header colour. 
Inherit the PWA tint from the mobile header on mobile devices to fill in the block around notches/dynamic islands on iPhones (checked in iOS simulator) 

image

Removed the corner rounding/outline in the search dropdown

### CODE REVIEW

**Comment by guise** [comment]
This is probably a good place to raise this suggestion… 

Thoughts on adding padding to the search box to make it's boundaries clearer, and coloured more like it's desktop appearance? @lowPolySkeleton @bhollis @robojumper

image

This could just be personal bias, but I really dislike material-design-esque input fields that are indicated exclusively with a 'single underline' — it's an objectively bad affordance. 

When the search box is full bleed into the margins it makes it harder to differentiate from other blocks of content (header, PWA bar etc) - they just look like stacked rectangles here… 
image

**Comment by bhollis** [comment]
Yeah I'm totally into redesigning the mobile search box. The main reason it's full bleed is to make as much use of the space as possible - even mildly complicated searches, plus all the buttons and icons, crowd out the actual input. I'd always intended to make the input grow as you add more text (up to a point)...

File: src/app/css-variables.ts [hunk 1]
```diff
@@ -80,7 +80,13 @@ export default function updateCSSVariables() {
  * Read the --theme-pwa-background CSS variable and use it to set the meta theme-color element.
  */
 export function syncThemeColor() {
-  const background = getComputedStyle(document.body).getPropertyValue('--theme-pwa-background');
+  let background = getComputedStyle(document.body).getPropertyValue('--theme-pwa-background');
+
+  // Extract tint from mobile header on mobile devices to match notch/dynamic island fill
+  if (/Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini/i.test(navigator.userAgent)) {
```
Comment (bhollis) [inline src/app/css-variables.ts:86]: Better to use our `isPhonePortrait` media query here instead of user agent sniffing.

**Comment by bhollis** [comment]
I made the `isPhonePortrait` tweak. I also fixed the shadow/border color behind the menu badge, and moved the color of upgrades back to orange - the red felt too much like an error state than the more friendly orange. I could be convinced to go with a green...

**Review by guise — COMMENTED** [review]
Awesome, thanks!

I'll raise a separate PR for the mobile search box design changes as that will probably warrant dedicated feedback/discussion

**Review by bhollis — APPROVED** [review]

**Pull request merged**